appmail/models.py

```diff
@@ -240,28 +250,36 @@ def create_message(self, context, **email_kwargs):
         these are set from the template (subject, body_text and body_html).
 
         """
-        for kw in ('subject', 'body', 'alternatives'):
-            assert kw not in email_kwargs, _lazy("Invalid create_message kwarg: '{}'".format(kw))
+        if "subject" in email_kwargs:
+            raise ValueError(_("Invalid argument: 'subject' is set from the template."))
+        if "body" in email_kwargs:
+            raise ValueError(_("Invalid argument: 'body' is set from the template."))
+        if "alternatives" in email_kwargs:
+            raise ValueError(
+                _("Invalid argument: 'alternatives' is set from the template.")
+            )
+        if email_kwargs.get("attachments", None) and not self.supports_attachments:
+            raise ValueError(_("Email template does not support attachments."))
```

Comment:
This is the content of the PR - everything else (beyond the definition of the field itself) is Black / isort / prettier.